File: backend/web/apps/product/api/views/brand/brand_detail.py
# apps/product/api/views/brand/brand_detail.py
from rest_framework.views import APIView
from rest_framework.response import Response
from rest_framework import status
from django.shortcuts import get_object_or_404
from django.core.paginator import Paginator
from apps.product.models.brand import Brand
from apps.product.api.serializers.brand.brand_product import (
    BrandDetailSerializer,
    BrandProductListSerializer,
    BrandCategoryListSerializer,
    BrandWithProductsFilterSerializer,
)
from apps.product.models.category import Category
from apps.core.api.response import success_response
from django.db import models

class BrandDetailView(APIView):
    """
    دریافت جزئیات برند با محصولات مرتبط و فیلتر دسته‌بندی

    پارامترهای URL:
    - brand_slug: اسلاگ برند

    پارامترهای Query:
    - category: اسلاگ دسته‌بندی (اختیاری)
    - page: شماره صفحه (پیش‌فرض: 1)
    - page_size: تعداد آیتم در صفحه (پیش‌فرض: 20)
    """

    def get(self, request, brand_slug, *args, **kwargs):
        # ============================================================
        # 1. دریافت برند
        # ============================================================
        brand = get_object_or_404(
            Brand.objects.filter(is_active=True),
            slug=brand_slug
        )

        # ============================================================
        # 2. دریافت پارامترهای Query
        # ============================================================
        category_slug = request.query_params.get('category')
        page = int(request.query_params.get('page', 1))
        page_size = int(request.query_params.get('page_size', 20))

        # محدود کردن page_size
        if page_size > 100:
            page_size = 100

        # ============================================================
        # 3. کوئری پایه محصولات
        # ============================================================
        products_queryset = brand.products.filter(
            status='published'
        ).select_related('brand').prefetch_related('categories', 'sales')

        # ============================================================
        # 4. اعمال فیلتر دسته‌بندی
        # ============================================================
        if category_slug:
            products_queryset = products_queryset.filter(
                categories__slug=category_slug,
                categories__status='published'
            )

        # ============================================================
        # 5. دریافت دسته‌بندی‌های مرتبط با برند
        # ============================================================
        categories = (
            Category.objects
            .filter(
                products__brand=brand,
                products__status='published',
                status='published'
            )
            .annotate(
                product_count=models.Count('products')
            )
            .filter(
                product_count__gt=0
            )
            .order_by('-product_count', 'title')
        )

        # ============================================================
        # 6. صفحه‌بندی محصولات
        # ============================================================
        paginator = Paginator(products_queryset, page_size)
        page_obj = paginator.get_page(page)

        # ============================================================
        # 7. سریالایزرها
        # ============================================================
        # سریالایزر برند
        brand_serializer = BrandDetailSerializer(
            brand,
            context={'request': request}
        )

        # سریالایزر محصولات
        products_serializer = BrandProductListSerializer(
            page_obj,
            many=True,
            context={'request': request}
        )

        # سریالایزر دسته‌بندی‌ها
        categories_serializer = BrandCategoryListSerializer(
            categories,
            many=True,
            context={'request': request}
        )

        # ============================================================
        # 8. پاسخ نهایی
        # ============================================================
        response_data = {
            'brand': brand_serializer.data,
            'categories': categories_serializer.data,
            'products': products_serializer.data,
            'pagination': {
                'total_count': paginator.count,
                'page': page,
                'page_size': page_size,
                'total_pages': paginator.num_pages,
                'has_next': page_obj.has_next(),
                'has_previous': page_obj.has_previous(),
            },
            'total_products': paginator.count,
        }

        return Response(
            success_response(data=response_data),
            status=status.HTTP_200_OK
        )

File: frontend/my-app/lib/api/brands/brandDetail.ts
// lib/api/brands/brandDetail.ts
import { apiGet } from "../client";

// ============================================================
// Types
// ============================================================
export interface BrandCategory {
  id: number;
  title: string;
  slug: string;
  image: string | null;
  product_count: number;
}

export interface BrandProduct {
  id: number;
  title: string;
  slug: string;
  cover_image_url: string | null;
  brand_name: string;
  brand_slug: string;
  categories: {
    id: number;
    title: string;
    slug: string;
  }[];
  price: string | null;
  min_price: string | null;
  max_price: string | null;
  status: string;
  created_at: string;
}

export interface BrandDetail {
  id: number;
  name: string;
  slug: string;
  logo_url: string | null;
  description: string;
  product_count: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface BrandDetailResponse {
  success: boolean;
  data: {
    brand: BrandDetail;
    categories: BrandCategory[];
    products: BrandProduct[];
    pagination: {
      total_count: number;
      page: number;
      page_size: number;
      total_pages: number;
      has_next: boolean;
      has_previous: boolean;
    };
    total_products: number;
  };
}

export interface BrandFilterParams {
  category?: string;
  page?: number;
  page_size?: number;
}

// ============================================================
// Main Function
// ============================================================
export async function getBrandDetail(
  brandSlug: string,
  params: BrandFilterParams = {}
): Promise<BrandDetailResponse['data']> {
  const queryParams = new URLSearchParams();

  if (params.category) queryParams.append('category', params.category);
  if (params.page) queryParams.append('page', String(params.page));
  if (params.page_size) queryParams.append('page_size', String(params.page_size));

  const queryString = queryParams.toString();
  const url = `/api/v1/brand_detail/${brandSlug}/?${queryString}`;

  console.log('🔥 Fetching Brand URL:', url);

  const response = await apiGet<BrandDetailResponse>(url);

  if (!response.success) {
    throw new Error('دریافت برند ناموفق بود.');
  }

  return response.data;
}

export async function getBrandList(params?: { limit?: number; search?: string }) {
  const queryParams = new URLSearchParams();

  if (params?.limit) queryParams.append('limit', String(params.limit));
  if (params?.search) queryParams.append('search', params.search);

  const queryString = queryParams.toString();
  const url = `/api/v1/brand_detail/?${queryString}`;

  const response = await apiGet<{ success: boolean; data: BrandDetail[] }>(url);

  if (!response.success) {
    throw new Error('دریافت برندها ناموفق بود.');
  }

  return response.data;
}